### ISSUE TITLE
Multiple recipes reported for local repo of org

### ISSUE THREAD

**Issue by matthuszagh** [raxod502/straight.el#582]
I'm using the following straight recipe for org-mode:

```elisp
(straight-use-package '(org :local-repo "~/src/org-mode"))
```

However, this fails with the following warning:
```
Warning (straight): Two different recipes given for "org" (:local-repo cannot be both "org" and "~/src/org-mode")
```

Have I somehow misused the straight recipe syntax? How can I install org-mode from a local directory?

I'm using the latest straight commit (5d046ad09413f3aba6198229e392cdd3f07e77ee) and a very recent Emacs commit.

Thanks for this great package! And let me know if this is missing any information that would be helpful.

**Comment by matthuszagh** [comment]
I realized I was unintentionally loading org prior to this as a dependency. However, when I resolved this, I'm now getting the error `Invalid version syntax: 'N/A'`. `straight-fix-org` is set to `t`.

**Comment by matthuszagh** [comment]
I was able to get this to work by explicitly setting
```
(setq straight-use-package-prepare-functions '(straight--fix-org-function))
```
before the `straight-use-package` call. Explicitly setting `(setq straight-fix-org t)` didn't work for me. This is strange, no? Shouldn't this be done by default? Is `:local-repo` not triggering the org hack? Or, have I done something wrong somewhere?

**Comment by progfolio** [comment]
If you're feeling adventurous, I have an open pull which properly builds Org:
https://github.com/raxod502/straight.el/pull/535

**Comment by matthuszagh** [comment]
Ok, it appears what I've actually encountered is [this issue with native compilation](https://debbugs.gnu.org/cgi/bugreport.cgi?bug=40838#29). I tried the fix there but didn't have any luck and have just decided to stick with non-native for the time being.

@progfolio that sounds like an great addition to straight! I unfortunately had some trouble with it, but I think maybe that was caused by the native compilation. I have some bandwidth to play around with this further and give it its fair due, I'll be sure to leave any feedback on the PR link. In any event, I'm closing this issue for now because it doesn't appear to be an issue with straight.
